src/components/icons/interfaces.icons.ts

```diff
@@ -3,11 +3,18 @@ import mongoose from 'mongoose'
 
 export type iconType = 'animated' | 'static'
 
+export interface SuggestedTagI {
+    _id: string,
+    tag: string,
+}
+
 // Interface of the icons objects stored in db:
 export interface IconInterface extends mongoose.Document{
     name: string,
     svg: string,
     tags: string[],
+    suggestedTags: SuggestedTagI,
```

Comment:
why are we keeping this in the same collection? shouldn't we have a separate collection for this feature?

1. Suppose people suggest 100k tags, this icon model collection is gonna get spammed heavily with data.
2. What if we add more features? like manual review + voting, we cant add everything here.

Reply:
Hmm, referencing or embedding both ways could work, we can use a separate collection for it, and reference the icon by either name or id, but this would require two requests to the db to retrieve all the icon's details ( since all the info, including the suggestedTags ) will be shown in the same place, if we normalized everything it would look like we're using a SQL db, Anyway, both solutions would work for us, 100k tag isn't that huge, for the voting and manual review, only two fields will be added to the `suggestedTagsI`, `votes`, `isApproved` which will be able to handle everything, as I've planned before cancelling the votes. But I do agree with the idea that if later we decided maybe to allow the user to suggest categories or whatever, the collection would be too large - it's not a real problem - but it can be solved if we followed the referencing approach, maybe we can create a collection `suggestions` with the following schema: 
```TS
interface SuggInerface {
type: tag | category | .... | ... |, 
iconName: string
suggestion: string
}
```

do you prefer going with a separate collection for this?

Reply:
what do you think @kbhutani0001

Reply:
Yes, this looks good. suggestion will be tag or category or like general string (like hey, i think you should add this category)?

Reply:
Awesome, I will work on it.

Reply:
should it be an array? i am thinking UX wise on FE we wouldn't want them to suggest only one tag. It should be comma separated or a list of checkbox they can select whichever they feel are relatable (we could need multiple inputs in future). You can keep it string and we can migrate the DB later or start with the array itself. What do you think?

Reply:
Yes, You're right, it would be better if we allowed them to suggest multiple tag/category in the same time, comma separated would be the best solution, and the backend will do the rest

Reply:
I think you should expect BE to get array from FE only, the rest should be managed there. on BE you cannot assume it will be a comma separated string or not and split it based on that as on FE we could be using same API for multiple things (tags, categories) so we can generate the payload accordingly.

Reply:
I was thinking of something like this: 
```JS
const mysuggest1 = 'fawzi,kartikay'
const mysuggest2 = 'fawzi'

console.log(mysuggest1.split(',')) // ["fawzi","kartikay"]
console.log(mysuggest2.split(',')) // ["fawzi"]
```
it would work for either ways, a single or multiple words
so the FE would send something like this: 
```JSON
type: "tag",
iconName: "abstract",
suggestion: "fawzi, kartikay, sorin"
```